sql/core/src/main/scala/org/apache/spark/sql/execution/command/AnalyzeTableCommand.scala

```diff
@@ -81,6 +83,19 @@ case class AnalyzeTableCommand(
 object AnalyzeTableCommand extends Logging {
 
   def calculateTotalSize(sessionState: SessionState, catalogTable: CatalogTable): Long = {
+    if (catalogTable.partitionColumnNames.isEmpty) {
+      calculateTotalSize(sessionState, catalogTable.identifier, catalogTable.storage.locationUri)
+    } else {
+      // Table = Sum(partitions)
+      val partitions = sessionState.catalog.listPartitions(catalogTable.identifier)
+      partitions.map(p =>
+        calculateTotalSize(sessionState, catalogTable.identifier, p.storage.locationUri)
+      ).sum
+    }
+  }
+
+  private def calculateTotalSize(sessionState: SessionState, tableId: TableIdentifier,
```

Comment:
nit: if the number of args exceeds line length, the code style is to have individual args on separate line with 4 spaces.

```
  private def calculateTotalSize(
      sessionState: SessionState,
      tableId: TableIdentifier,
      locationUri: Option[URI]): Long = {
    // This method is mainly based on
    ....
```